### PR TITLE
Python3 support for spiro

### PR DESCRIPTION
**Pull request by RaspberryPiFan**
This makes spiro comaptible to python3, but it should still work with python2

### CODE REVIEW

File: third_party/spiro/x3/pyrex/main.py [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
         dc.lineto(100, 100)
-        print dc.currentpoint()
+        print(dc.currentpoint())
         dc.stroke()
         dc.selectfont("Nimbus Sans L", 0, 0)
         dc.setfontsize(12)
         dc.moveto(50, 10)
         dc.showtext(u"\u00a1hello, world!")
-        print dc.textextents(u"\u00a1hello, world!")
+        print(dc.textextents(u"\u00a1hello, world!")(
```
Comment (whitslack) [inline third_party/spiro/x3/pyrex/main.py]:
```suggestion
        print(dc.textextents(u"\u00a1hello, world!"))
```

File: third_party/spiro/font/cut.py [hunk 2]
```diff
@@ -39,9 +39,9 @@
             ofn = pref + '%04d.pgm' % ix
             of = file(ofn, 'w')
             active[ix] = of
-            print >> of, 'P5'
-            print >> of, r[2] - r[0], r[3] - r[1]
-            print >> of, '255'
+            print(>> of, 'P5')
+            print(>> of, r[2] - r[0], r[3] - r[1)
```
Comment (whitslack) [inline third_party/spiro/font/cut.py]:
```suggestion
            print(>> of, r[2] - r[0], r[3] - r[1])
```

File: third_party/spiro/curves/bezfigs.py [hunk 3]
```diff
@@ -7,16 +7,16 @@
...
+    print('%%EndComments')
+    print('%%EndProlog')
+    print('%%Page: 1 1')
     if draw_box:
-        print x0, y0, 'moveto', x0, y1, 'lineto', x1, y1, 'lineto', x1, y0, 'lineto closepath stroke'
+        print(x0, y0, 'moveto', x0, y1, 'lineto', x1, y1, 'lineto', x1, y0, 'lineto closepath stroke')
 
 def eps_trailer():
-    print '%%EOF'
+    print('%%EOF'
```
Comment (whitslack) [inline third_party/spiro/curves/bezfigs.py]:
```suggestion
    print('%%EOF')
```

File: third_party/spiro/curves/bigmat.py [hunk 4]
```diff
@@ -196,7 +196,7 @@ def inversedot(m, v):
             u[i] = v[i % n]
             for j in range(-7, 4):
                 a[i, j + 7] = m[i % n, (i + j + 7 * n) % n]
-        #print a
+        #print(a
```
Comment (whitslack) [inline third_party/spiro/curves/bigmat.py]:
```suggestion
        #print(a)
```

File: third_party/spiro/curves/bigmat.py [hunk 5]
```diff
@@ -205,15 +205,15 @@ def inversedot(m, v):
                     elif a[i, j] < 0: sys.stdout.write('- ')
                     else: sys.stdout.write('  ')
                 sys.stdout.write('\n')
-        #print u
+        #print(u
         al, indx, d = band.bandec(a, 5, 5)
         band.banbks(a, 5, 5, al, indx, u)
-        #print u
+        #print(u
```
Comment (whitslack) [inline third_party/spiro/curves/bigmat.py]:
```suggestion
        #print(u)
```

File: third_party/spiro/curves/bigmat.py [hunk 6]
```diff
@@ -367,7 +367,7 @@ def iter(segs, nodes):
 
         j += jinc
         j0 = j1
-    #print m
+    #print(m
```
Comment (whitslack) [inline third_party/spiro/curves/bigmat.py]:
```suggestion
    #print(m)
```

File: third_party/spiro/curves/bigmat.py [hunk 7]
```diff
@@ -465,70 +465,70 @@ def plot_ks(segs, nodes, xo, yo, xscale, yscale):
...
 def prologue():
-    print '/ss 2 def'
-    print '/circle { ss 0 moveto currentpoint exch ss sub exch ss 0 360 arc } bind def'
-    print '/poly {'
-    print '    dup false exch {'
-    print '        0 3 index 2 index { lineto } { moveto } ifelse pop true'
-    print '        360.0 2 index div rotate'
-    print '    } repeat pop pop pop'
-    print '} bind def'
+    print('/ss 2 def'
```
Comment (whitslack) [inline third_party/spiro/curves/bigmat.py]:
```suggestion
    print('/ss 2 def')
```

File: third_party/spiro/curves/bigmat.py [hunk 8]
```diff
@@ -465,70 +465,70 @@ def plot_ks(segs, nodes, xo, yo, xscale, yscale):
...
+    print('/circle { ss 0 moveto currentpoint exch ss sub exch ss 0 360 arc } bind def'
```
Comment (whitslack) [inline third_party/spiro/curves/bigmat.py]:
```suggestion
    print('/circle { ss 0 moveto currentpoint exch ss sub exch ss 0 360 arc } bind def')
```

File: third_party/spiro/curves/bigmat.py [hunk 9]
```diff
@@ -465,70 +465,70 @@ def plot_ks(segs, nodes, xo, yo, xscale, yscale):
...
+    print('/poly {'
```
Comment (whitslack) [inline third_party/spiro/curves/bigmat.py]:
```suggestion
    print('/poly {')
```

File: third_party/spiro/curves/bigmat.py [hunk 10]
```diff
@@ -465,70 +465,70 @@ def plot_ks(segs, nodes, xo, yo, xscale, yscale):
...
+    print('    dup false exch {'
```
Comment (whitslack) [inline third_party/spiro/curves/bigmat.py]:
```suggestion
    print('    dup false exch {')
```

File: third_party/spiro/curves/bigmat.py [hunk 11]
```diff
@@ -465,70 +465,70 @@ def plot_ks(segs, nodes, xo, yo, xscale, yscale):
...
+    print('        0 3 index 2 index { lineto } { moveto } ifelse pop true'
```
Comment (whitslack) [inline third_party/spiro/curves/bigmat.py]:
```suggestion
    print('        0 3 index 2 index { lineto } { moveto } ifelse pop true')
```

File: third_party/spiro/curves/bigmat.py [hunk 12]
```diff
@@ -465,70 +465,70 @@ def plot_ks(segs, nodes, xo, yo, xscale, yscale):
...
+    print('        360.0 2 index div rotate'
```
Comment (whitslack) [inline third_party/spiro/curves/bigmat.py]:
```suggestion
    print('        360.0 2 index div rotate')
```

File: third_party/spiro/curves/bigmat.py [hunk 13]
```diff
@@ -465,70 +465,70 @@ def plot_ks(segs, nodes, xo, yo, xscale, yscale):
...
+    print('    } repeat pop pop pop'
```
Comment (whitslack) [inline third_party/spiro/curves/bigmat.py]:
```suggestion
    print('    } repeat pop pop pop')
```

File: third_party/spiro/curves/bigmat.py [hunk 14]
```diff
@@ -465,70 +465,70 @@ def plot_ks(segs, nodes, xo, yo, xscale, yscale):
...
+    print('} bind def'
```
Comment (whitslack) [inline third_party/spiro/curves/bigmat.py]:
```suggestion
    print('} bind def')
```

File: third_party/spiro/curves/bigmat.py [hunk 15]
```diff
@@ -465,70 +465,70 @@ def plot_ks(segs, nodes, xo, yo, xscale, yscale):
...
 
 def run_path(path, show_iter = False, n = 10, xo = 36, yo = 550, xscale = .25, yscale = 2000, pl_nodes = True):
     segs, nodes = setup_path(path)
-    print '.5 setlinewidth'
+    print('.5 setlinewidth'
```
Comment (whitslack) [inline third_party/spiro/curves/bigmat.py]:
```suggestion
    print('.5 setlinewidth')
```

File: third_party/spiro/curves/bigmat.py [hunk 16]
```diff
@@ -465,70 +465,70 @@ def plot_ks(segs, nodes, xo, yo, xscale, yscale):
...
     for i in range(n):
         if i == n - 1:
-            print '0 0 0 setrgbcolor 1 setlinewidth'
+            print('0 0 0 setrgbcolor 1 setlinewidth'
```
Comment (whitslack) [inline third_party/spiro/curves/bigmat.py]:
```suggestion
            print('0 0 0 setrgbcolor 1 setlinewidth')
```

File: third_party/spiro/curves/bigmat.py [hunk 17]
```diff
@@ -465,70 +465,70 @@ def plot_ks(segs, nodes, xo, yo, xscale, yscale):
...
         elif i == 0:
-            print '1 0 0 setrgbcolor'
+            print('1 0 0 setrgbcolor'
```
Comment (whitslack) [inline third_party/spiro/curves/bigmat.py]:
```suggestion
            print('1 0 0 setrgbcolor')
```

File: third_party/spiro/curves/bigmat.py [hunk 18]
```diff
@@ -465,70 +465,70 @@ def plot_ks(segs, nodes, xo, yo, xscale, yscale):
...
         elif i == 1:
-            print '0 0.5 0 setrgbcolor'
+            print('0 0.5 0 setrgbcolor'
```
Comment (whitslack) [inline third_party/spiro/curves/bigmat.py]:
```suggestion
            print('0 0.5 0 setrgbcolor')
```

File: third_party/spiro/curves/bigmat.py [hunk 19]
```diff
@@ -465,70 +465,70 @@ def plot_ks(segs, nodes, xo, yo, xscale, yscale):
...
         elif i == 2:
-            print '0.3 0.3 0.3 setrgbcolor'
+            print('0.3 0.3 0.3 setrgbcolor'
```
Comment (whitslack) [inline third_party/spiro/curves/bigmat.py]:
```suggestion
            print('0.3 0.3 0.3 setrgbcolor')
```

File: third_party/spiro/curves/bigmat.py [hunk 20]
```diff
@@ -465,70 +465,70 @@ def plot_ks(segs, nodes, xo, yo, xscale, yscale):
...
         norm = iter(segs, nodes)
-        print '% norm =', norm
+        print('% norm =', norm
```
Comment (whitslack) [inline third_party/spiro/curves/bigmat.py]:
```suggestion
        print('% norm =', norm)
```

File: third_party/spiro/curves/euler-elastica.py [hunk 21]
```diff
@@ -12,18 +12,18 @@ def plot_elastica(a, c):
         x = .001
     try:
         for i in range(1000):
-            print 6 + s * x, 200 + s * y, cmd
+            print(6 + s * x, 200 + s * y, cmd
```
Comment (whitslack) [inline third_party/spiro/curves/euler-elastica.py]:
```suggestion
            print(6 + s * x, 200 + s * y, cmd)
```

File: third_party/spiro/curves/fromcubic.py [hunk 22]
```diff
@@ -14,7 +14,7 @@ def read_bz(f):
         s = l.split()
         if len(s) > 0:
             cmd = s[-1]
-            #print s[:-1], cmd
+            #print(s[:-1], cmd
```
Comment (whitslack) [inline third_party/spiro/curves/fromcubic.py]:
```suggestion
            #print(s[:-1], cmd)
```

File: third_party/spiro/curves/fromcubic.py [hunk 23]
```diff
@@ -37,24 +37,24 @@ def plot_bzs(bzs, z0, scale, fancy = False):
         for i in range(len(sp)):
             bz = sp[i]
             tocubic.plot_bz(bz, z0, scale, i == 0)
-        print 'stroke'
+        print('stroke'
```
Comment (whitslack) [inline third_party/spiro/curves/fromcubic.py]:
```suggestion
        print('stroke')
```

File: third_party/spiro/curves/fromcubic.py [hunk 24]
```diff
@@ -68,7 +68,7 @@ def measure_derivs(x, ys):
         ds = hypot(dx, dy)
         s = ys[0] * arclen_ratio
         dscore = ds * (tocubic.mod_2pi(atan2(dy, dx) - seg.th(s)) ** 2)
-        #print s, atan2(dy, dx),  seg.th(s)
+        #print(s, atan2(dy, dx),  seg.th(s)
```
Comment (whitslack) [inline third_party/spiro/curves/fromcubic.py]:
```suggestion
        #print(s, atan2(dy, dx),  seg.th(s))
```

File: third_party/spiro/curves/fromcubic.py [hunk 25]
```diff
@@ -144,65 +144,65 @@ def test_to_pcorn():
...
-        print '0 20 translate'
+        print(>> sys.stderr, thresh, len(segs))
+        print('0 20 translate')
 
 if __name__ == '__main__':
     f = file(sys.argv[1])
     bzs = read_bz(f)
     rsps = bzs_to_pcorn(bzs, 1)
-    #print rsps
+    #print(rsps
```
Comment (whitslack) [inline third_party/spiro/curves/fromcubic.py]:
```suggestion
    #print(rsps)
```

File: third_party/spiro/curves/fromcubic.py [hunk 26]
```diff
@@ -144,65 +144,65 @@ def test_to_pcorn():
...
+            print('gsave 1 0 0 setrgbcolor')
             cmd = 'moveto'
             for i in range(100):
                 s = i * .01 * curve.arclen
                 x, y = curve.xy(s)
                 th = curve.th(s)
                 sth = 5 * sin(th)
                 cth = 5 * cos(th)
-                print x, y, cmd
+                print(x, y, cmd
```
Comment (whitslack) [inline third_party/spiro/curves/fromcubic.py]:
```suggestion
                print(x, y, cmd)
```

**Comment by RaspberryPiFan** [comment]
I just forgot that file (third_party/spiro/curves/bigmat.py) I'll create a new PR.

**Comment by marekjez86** [comment]
@whitslack 
@RaspberryPiFan 
I cannot merge this branch unless both of you sign Contributor License Agreement at https://cla.developers.google.com/ ... Could you to do it?

**Comment by whitslack** [comment]
> I cannot merge this branch unless both of you sign Contributor License Agreement

@marekjez86: Submitted.

**Comment by RaspberryPiFan2** [comment]
I have submitted it, but I currently can't login into my accountto do more fixes on this PR

**Comment by RaspberryPiFan2** [comment]
@googlebot I fixed it.

**Pull request closed without merging**